tools/ci/tc/decision.py

```diff
@@ -126,6 +124,7 @@ def filter_schedule_if(event, tasks):
                     scheduled[name] = task
         else:
             scheduled[name] = task
+    filter_tasks(scheduled, event)
```

Comment:
Instead of adding this to `filter_schedule_if()` can we add this to `decide()` after `filter_schedule_if()`? The logging here will still indicate the tasks that were added from the yaml definitions, so we should probably also add a log in `filter_tasks()` to indicate that some tasks were filtered.

Reply:
That is a definitely a more organized approach - I reimplemented it this way!